### PR TITLE
V2.2

### PR DESCRIPTION
**Pull request by bnlawrence**
I've broken the various pieces of the V2.2. into package sized commits. The original commits can still be seen in the master of my fork (I'll move them, but keep them).

This is marked as v2.2pre. I think it would be ok to accept this pull request for now, and make a more considered set of decisions when pulling into master (after tagging the current version into it's own branch for ongoing support). Suggest we discuss all the detail of this anon.

### CODE REVIEW

**Comment by bnlawrence** [comment]
The related dataset formal.association is broken. shared.formal_association doesn't exist.
(This is also a pyosl issue, why wasn't that picked up in the metamodel check?)

File: platform_classes.py [hunk 1]
```diff
@@ -3,38 +3,11 @@
 """
 .. module:: platform_classes.py
    :synopsis: Set of CIM v2 ontology type definitions.
+   applying to the description of computers and the
+   performance of code running on those computers. of
```
Comment (sadielbartholomew) [inline platform_classes.py:7]:
```suggestion
   performance of code running on those computers.
```

File: platform_classes.py [hunk 2]
```diff
@@ -80,6 +53,42 @@ def compute_pool():
     }
 
 
+def interconnect():
+    """ The interconnect used within a machine to joining nodes together"""
```
Comment (sadielbartholomew) [inline platform_classes.py]:
```suggestion
    """ The interconnect used within a machine to join nodes together"""
```

Comment (sadielbartholomew) [reply platform_classes.py]: Or 'for joining', I suppose. I imagine one of the two was intended.

File: platform_classes.py [hunk 3]
```diff
@@ -80,6 +53,42 @@ def compute_pool():
...
+            ('name', 'str', '0.1',
+                "Name of interconnnect."),
+            ('topology', 'str', '0.1',
+                'Interconnect topology'),
+            ('description', 'str', '0.1.',
+                'Technical description of interconnect layout'),
+            ('vendor', 'linked_to(shared.party)', '0.1',
+                'Supplier of the interconnect')
+            ]
+            }
```
Comment (sadielbartholomew) [inline platform_classes.py:73]:
```suggestion
    }
```

**Review by sadielbartholomew — COMMENTED** [review]
Hi Bryan, thought I'd register a small number of typos and formatting issues in the `platform_classes` module I noticed when looking at it in detail as part of the Machine and Performance work.

We'll try to catch you to discuss getting this PR merged soon. Thanks.

**Review by asladeofgreen — APPROVED** [review]
@sadielbartholomew and @momipsl have reviewed and are content that this change set achieves the objectives it was set out do to so.  We will double check that post commit the ontology will be functional from a pragmatic perspective.

**Pull request merged**